### PR TITLE
npm-audit

### LINKED ISSUE #31: Port npm-audit to reviewdog

### PR DESCRIPTION
**Pull request by bcaller**
Hooray for nonsense ReDoS findings

Closes #31

### CODE REVIEW

**Comment by bcaller** [comment]
I hate trying to debug this.

File: t3sts/npmaudit/package-lock.json [hunk 1]
```diff
@@ -0,0 +1,7394 @@
...
+      "dependencies": {
+        "bindings": "^1.5.0",
+        "nan": "^2.12.1"
+      },
+      "engines": {
+        "node": ">= 4.0"
+      }
+    },
+    "node_modules/watchpack-chokidar2/node_modules/glob-parent": {
+      "version": "3.1.0",
```
[the remark this answers is not in the document]
Comment (diracdeltas) [reply t3sts/npmaudit/package-lock.json:3864]: lol i love it

@bcaller maybe add a rule to exclude it if the audit description includes regex dos (preferably using a regex that itself is vulnerable to regex dos)

**Comment by thypon** [comment]
Wonder if it's possible to add an advised remediation in the text, and the chain of deps that brought the vulnlib in.

No rush here, we can decide to either merge and track the advises or implement them here

**Review by thypon — APPROVED** [review]

**Comment by thypon** [comment]
Can/should we change name to `npm-audit`?

**Comment by bcaller** [comment]
> Wonder if it's possible to add an advised remediation in the text, and the chain of deps that brought the vulnlib in.
> 
> No rush here, we can decide to either merge and track the advises or implement them here

It's possible. For now the highlighted line shows part of the dependency tree which should help.

File: t3sts/npmaudit/package-lock.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (bcaller) [reply t3sts/npmaudit/package-lock.json:3864]: I'll leave it for now :) but if we want later we can filter out CWE-400.

**Pull request merged**